spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/AbstractHazelcastMessageProducer.java

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.EventObject;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.hazelcast.core.AbstractIMapEvent;
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.EntryEvent;
+import com.hazelcast.core.EntryListener;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.MapEvent;
+import com.hazelcast.core.MultiMap;
+
+import reactor.util.StringUtils;
+
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.hazelcast.common.CacheEventType;
+import org.springframework.integration.hazelcast.common.CacheListeningPolicyType;
+import org.springframework.integration.hazelcast.common.HazelcastIntegrationDefinitionValidator;
+import org.springframework.integration.hazelcast.config.HazelcastLocalInstanceRegistrar;
+import org.springframework.util.Assert;
+
+/**
+ * Hazelcast Base Event-Driven Message Producer.
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+public abstract class AbstractHazelcastMessageProducer extends MessageProducerSupport {
+
+	private final DistributedObject distributedObject;
+
+	private String cacheEventTypes;
+
+	private CacheListeningPolicyType cacheListeningPolicy;
+
+	private String hazelcastRegisteredEventListenerId;
+
+	private Set<String> cacheEvents;
+
+	protected AbstractHazelcastMessageProducer(DistributedObject distributedObject) {
+		Assert.notNull(distributedObject, "cache must not be null");
+		this.distributedObject = distributedObject;
+	}
+
+	protected Set<String> getCacheEvents() {
+		return cacheEvents;
+	}
+
+	protected DistributedObject getDistributedObject() {
+		return distributedObject;
+	}
+
+	public void setCacheEventTypes(String cacheEventTypes) {
+		HazelcastIntegrationDefinitionValidator.validateEnumType(CacheEventType.class, cacheEventTypes);
+		this.cacheEventTypes = cacheEventTypes;
+	}
+
+	protected CacheListeningPolicyType getCacheListeningPolicy() {
+		return cacheListeningPolicy;
+	}
+
+	public void setCacheListeningPolicy(CacheListeningPolicyType cacheListeningPolicy) {
+		this.cacheListeningPolicy = cacheListeningPolicy;
+	}
+
+	protected String getHazelcastRegisteredEventListenerId() {
+		return hazelcastRegisteredEventListenerId;
+	}
+
+	protected void setHazelcastRegisteredEventListenerId(String hazelcastRegisteredEventListenerId) {
+		this.hazelcastRegisteredEventListenerId = hazelcastRegisteredEventListenerId;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		this.cacheEvents = StringUtils.commaDelimitedListToSet(this.cacheEventTypes);
+		HazelcastIntegrationDefinitionValidator.validateCacheEventsByDistributedObject(
+				getDistributedObject(), this.cacheEvents);
```

Comment:
Looks like this code can be moved to the `setCacheEventTypes` and we can remove `cacheEventTypes` variable.
But here, I guess, we must have some `Assert` to be sure that `cacheEvents` isn't empty.
WDYT?